src/lightning/pytorch/callbacks/model_checkpoint.py

```diff
@@ -703,7 +703,7 @@ def _save_monitor_checkpoint(self, trainer: "pl.Trainer", monitor_candidates: Di
             rank_zero_info(f"Epoch {epoch:d}, global step {step:d}: {self.monitor!r} was not in top {self.save_top_k}")
 
     def _save_none_monitor_checkpoint(self, trainer: "pl.Trainer", monitor_candidates: Dict[str, Tensor]) -> None:
-        filepath = self._get_metric_interpolated_filepath_name(monitor_candidates, trainer)
+        filepath = self._get_metric_interpolated_filepath_name(monitor_candidates, trainer, self.best_model_path)
```

Comment:
`del_filepath` in `_get_metric_interpolated_filepath_name` could now be required to avoid this mistake. I don't remember why it was optional.

Reply:
The other place where del_filepath is used is here:
https://github.com/Lightning-AI/lightning/blob/e30401af80ba777459d0ab6b948ac5814fa34ae5/src/lightning/pytorch/callbacks/model_checkpoint.py#L719-L729

But I didn't see an obvious way to define it as not None, so it seems we still need it optional.